bob/src/hw_metrics_collector.rs

```diff
@@ -133,10 +149,168 @@ impl HWMetricsCollector {
             }
         }
 
-        DisksMetrics {
+        DisksSpaceMetrics {
             total_space: total,
             used_space: used,
-            free_space: free
+            free_space: free,
+        }
+    }
+}
+
+struct CPUStatCollector {
+    iostat_avl: bool,
+}
+
+impl CPUStatCollector {
+    fn new() -> CPUStatCollector {
+        CPUStatCollector { iostat_avl: true }
+    }
+
+    fn parse_iostat_result(iostat_str: &str) -> Result<f64, std::num::ParseFloatError> {
+        iostat_str.replace(',', ".").parse()
+    }
+
+    fn iowait(&mut self) -> Result<f64, String> {
+        if !self.iostat_avl {
+            return Err("iostat is not available".to_string());
+        }
+
+        let ios = std::process::Command::new("iostat").arg("-c").output();
+        match parse_command_output(ios) {
+            Ok(output) => {
+                // find avg-cpu headers line
+                let mut lines = output
+                    .lines()
+                    .skip_while(|line| line.find("avg-cpu:").is_none());
+                // find iowait column on the next line
+                if let Some(line) = lines.next() {
+                    let mut values = line.split_whitespace();
+                    // iowait is in 3d column
+                    if let Some(iowait_str) = values.nth(3) {
+                        if let Ok(iowait) = Self::parse_iostat_result(iowait_str) {
+                            return Ok(iowait);
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                self.iostat_avl = false;
+                return Err(e);
+            }
+        }
+        self.iostat_avl = false;
+        Err("Failed to get iowait from iostat".to_string())
```

Comment:
You should log errors after which `iostat_avl` is set to false. Probably, the best way to do that is to return not just a `String` for errors, but some Enum, indicating the reason of error, and process the return codes on the caller side

Reply:
This is logged in task() too

Reply:
The problem is that you are now logging `iostat is not available` on every attempt, which is what I wanted to avoid

Reply:
You can return an Enum as an error to distinguish between primary error and unavailability messages